Client/Frontend/Browser/Tab.swift

```diff
@@ -492,6 +492,18 @@ class Tab: NSObject {
         checkTabCount(failures: 0)
         #endif
     }
+    
+    /// When a user clears ALL history, `sessionData` and `historyList` need to be purged, and close the webView.
+    func clearAndResetTabHistory() {
+        guard let currentlyOpenUrl = sessionData?.urls.last ?? historyList.last ?? url else { return }
```

Comment:
Any particular reason we use session data last url instead of looking into ` into `lastKnownUrl` property on [here](https://github.com/mozilla-mobile/firefox-ios/blob/7f04e018925d5fd44b54367358c56dedff341320/Client/Frontend/Browser/Tab.swift#L185-L192) and just compare it with history list last url

Reply:
No particular reason. Would it be better to do it that way?

Reply:
It depends on the chain in your case do you want to check session data first? then history and lastly the url?

with `lastKnownUrl` it'll check `sessionData?.urls.last` > `url` > and lastly you can > `?? historyList.last`

if you want to check `historyList.last` in the middle then leave it the way it is. Happy to discuss more

Reply:
Ahh I see what you mean. Assuming lastKnownUrl can give us the currently open URL as well, I think making the check go this way makes sense:
`lastKnownURL --> historyList.last`